prow/plugins/cat/cat_test.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yuks
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+type fakeClowder string
+
+var human = flag.Bool("human", false, "Enable to run additional manual tests")
+var category = flag.String("category", "", "Request a particular category if set")
+
+func (c fakeClowder) readCat(category string) (string, error) {
+	if category == "error" {
+		return "", errors.New(string(c))
+	}
+	return string(c), nil
+}
+
+func TestRealCat(t *testing.T) {
+	if !*human {
+		t.Skip("Real cats disabled for automation. Manual users can add --human")
+	}
+	if cat, err := catURL.readCat(*category); err != nil {
+		t.Errorf("Could not read cats from %s: %v", catURL, err)
+	} else {
+		fmt.Println(cat)
+	}
+}
+
+func TestFormat(t *testing.T) {
+	re := regexp.MustCompile(`\[!\[.+\]\(.+\)\]\(.+\)`)
+	basicUrl := "http://example.com"
+	testcases := []struct {
+		name string
+		src  string
+		img  string
+		err  bool
+	}{
+		{
+			name: "basically works",
+			src:  basicUrl,
+			img:  basicUrl,
+			err:  false,
+		},
+		{
+			name: "empty source",
+			src:  "",
+			img:  basicUrl,
+			err:  true,
+		},
+		{
+			name: "empty image",
+			src:  basicUrl,
+			img:  "",
+			err:  true,
+		},
+		{
+			name: "bad source",
+			src:  "http://this is not a url",
+			img:  basicUrl,
+			err:  true,
+		},
+		{
+			name: "bad image",
+			src:  basicUrl,
+			img:  "http://still a bad url",
+			err:  true,
+		},
+	}
+	for _, tc := range testcases {
+		ret, err := catResult{
+			Source: tc.src,
+			Image:  tc.img,
+		}.Format()
+		switch {
+		case tc.err:
+			if err == nil {
+				t.Errorf("%s: failed to raise an error", tc.name)
+			}
+		case err != nil:
+			t.Errorf("%s: unexpected error: %v", tc.name, err)
+		case !re.MatchString(ret):
+			t.Errorf("%s: bad return value: %s", tc.name, ret)
+		}
+	}
+}
+
+// Medium integration test (depends on ability to open a TCP port)
+func TestJokesMedium(t *testing.T) {
```

Comment:
seems like this should have a different name

Reply:
Done